float32 min_height
float32 max_height
float32 current_height
uint8 error_code

